llama_index/evaluation/guideline_eval.py

```diff
@@ -0,0 +1,78 @@
+import logging
+from typing import Optional
+
+from langchain.output_parsers import PydanticOutputParser
+from pydantic import BaseModel, Field
+
+from llama_index.evaluation.base import BaseEvaluator, Evaluation
+from llama_index.indices.base import ServiceContext
+from llama_index.prompts.base import Prompt
+from llama_index.response.schema import Response
+
+logger = logging.getLogger(__name__)
+
+
+class GuidelineEvaluator(BaseEvaluator):
+    """An evaluator which uses guidelines to evaluate a response.
+
+    Args:
+        service_context(ServiceContext): The service context to use for evaluation.
+        guidelines(str): User-added guidelines to use for evaluation.
+    """
+
+    def __init__(
+        self,
+        service_context: Optional[ServiceContext] = None,
+        guidelines: Optional[str] = None,
+        eval_template: Optional[str] = None,
+    ) -> None:
+        self.service_context = service_context or ServiceContext.from_defaults()
+        self.guidelines = guidelines or DEFAULT_GUIDELINES
+        self.eval_template = eval_template or DEFAULT_EVAL_TEMPLATE
+
+    def evaluate_response(self, query: str, response: Response) -> Evaluation:
```

Comment:
curious: why use `Response` instead of a simpler `str` type in the interface. Do we expect to rely on other information like source nodes?

Reply:
This is based on BaseEvaluator.evaluate_response which keeps the Response param for compatibility with previous instantiations. You can see where the Response source nodes are used in QueryResponseEvaluator

Reply:
Gotcha, this makes sense.